utils/src/storage.rs

```diff
@@ -0,0 +1,137 @@
+use std::fs::{DirBuilder, File, ReadDir};
+use std::io::{Read, Write};
+use std::path::{Path, PathBuf};
+
+use bincode;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug)]
+pub struct FsStorageHelper {
+    root: PathBuf,
+}
+
+impl FsStorageHelper {
+    pub fn new(root_path: &Path) -> Result<Self, std::io::Error> {
+        DirBuilder::new()
+            .recursive(true)
+            .create(root_path)
+            .map(|_| Self {
+                root: root_path.to_path_buf(),
+            })
+    }
+
+    fn get_root_path(&self, path: Option<&Path>) -> PathBuf {
+        match path {
+            Some(p) => p.to_path_buf(),
```

Comment:
How does this work?

Reply:
If Some you actually just want what is inside, if None then you want the root.